SCSH_3C_iOS/Persistence/CoreData/Board/BoardCoreDataService.swift

```diff
@@ -0,0 +1,152 @@
+}
+
+public class BoardCoreDataService: BoardCoreDataServiceType {
+    
+    public init() {}
+    
+    public func get(code: String?) -> AnyPublisher<[Board], Error> {
+        Future<[Board], Error> { [weak self] promise in
+            guard let self = self else { return }
+            
+            promise(self.get(code: code))
+        }
+        .eraseToAnyPublisher()
+    }
+
+    public func save(board: Board) -> AnyPublisher<Void, Error> {
+        Future<Void, Error> { [weak self] promise in
+            guard let self = self else { return }
+            
+            promise(self.save(board: board))
+        }
+        .eraseToAnyPublisher()
+    }
+
+    public func saveAll(boards: [Board]) -> AnyPublisher<Void, Error> {
+        Future<Void, Error> { [weak self] promise in
+            guard let self = self else { return }
+            
+            promise(self.saveAll(boards: boards))
+        }
+        .eraseToAnyPublisher()
+    }
+    
+    private func get(code: String?, context: NSManagedObjectContext = CoreDataController.shared.container.newBackgroundContext()) -> Result<[Board], Error> {
+        let fetchRequest = Board.fetchRequest()
+        fetchRequest.sortDescriptors = [NSSortDescriptor(key: "id", ascending: true)]
+        
+        if let code = code {
+            fetchRequest.predicate = NSPredicate(format: "code == %@", code)
+        }
+        
+        do {
+            let results = try context.fetch(fetchRequest)
+            try context.save()
+            
+            return .success(results)
+            
+        } catch {
+            return .failure(error)
+        }
+    }
+    
+    private func save(board: Board) -> Result<Void, Error> {
+        guard let code = board.code else {
+            // TODO: Error handling
+            return .failure(NSError(domain: "", code: 0))
+        }
+        let context = board.managedObjectContext ?? CoreDataController.shared.container.newBackgroundContext()
+        let result: Result<[Board], Error> = get(code: code)
+
+        switch result {
+        case .success(let existingBoards):
+            
+            if let existingBoard = existingBoards.first {
+                update(existingBoard: existingBoard, newBoard: board)
+                context.delete(board)
+            } else {
+                context.insert(board)
+            }
+
+            do {
+                try context.save()
+
+                return .success(())
+            } catch {
+                print("❌❌❌ BoardCoreDataService saveAll save fail !")
+                print(error.localizedDescription)
+
+                return .failure(error)
+            }
+            
+        case .failure(let error):
+            print("❌❌❌ BoardCoreDataService saveAll get fail !")
+            print(error.localizedDescription)
+            
+            return .failure(error)
+        }
+    }
+    
+    private func update(existingBoard: Board, newBoard: Board) {
```

Comment:
The `update` method in the `BoardCoreDataService` already feels semantically clear enough, so IMHO the variable names for `board` are a bit redundant. 
```suggestion
    private func update(existing: Board, new: Board) {
```

Reply:
Agree !

Reply:
https://github.com/No-57/SCSH_3C_iOS/pull/45/commits/5e389baa626871767c8f5f86f95bb4224dd5a99d